pkg/azuredx/azureauth/aad_client.go

```diff
@@ -0,0 +1,69 @@
+package azureauth
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+	"github.com/grafana/grafana-azure-sdk-go/azcredentials"
+	"github.com/grafana/grafana-azure-sdk-go/azsettings"
+)
+
+// Abstraction over confidential.Client from MSAL for Go
+type aadClient interface {
+	AcquireTokenOnBehalfOf(ctx context.Context, userAssertion string, scopes []string) (confidential.AuthResult, error)
+}
+
+func newAADClient(credentials *azcredentials.AzureClientSecretCredentials, httpClient *http.Client) (aadClient, error) {
+	authority, err := resolveAuthorityForCloud(credentials.AzureCloud)
+	if err != nil {
+		return nil, fmt.Errorf("invalid Azure credentials: %w", err)
+	}
+
+	if !validTenantId(credentials.TenantId) {
+		return nil, errors.New("invalid tenantId")
+	}
+
+	clientCredential, err := confidential.NewCredFromSecret(credentials.ClientSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	authorityOpts := confidential.WithAuthority(runtime.JoinPaths(string(authority), credentials.TenantId))
+	httpClientOpts := confidential.WithHTTPClient(httpClient)
+
+	client, err := confidential.New(credentials.ClientId, clientCredential, authorityOpts, httpClientOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func resolveAuthorityForCloud(cloudName string) (azidentity.AuthorityHost, error) {
```

Comment:
We already have this func in the SDK, no? Can we use it instead?

Reply:
The end goal is to move everything from here to the SDK (`AzureTokenProvider`), so I don't want to expose this API in SDK as it won't be needed in future.
This is only incremental change. Here I just want to get read of home made implementation of token client to have `auth.go` cleaner for further refactoring.

Reply:
I'm fine with that!